### ISSUE TITLE
Install command causes "The folder contains an invalid version"/"Unable to find package" warnings/errors

### ISSUE THREAD

**Issue by mrahhal** [dotnet/templating#1287]
We have a web application template that we built on sdk 1.0 and it used to work great. After updating our tooling to 2.0 (but we still use aspnetcore 1.0), `dotnet new -i` started giving all sorts of warnings and errors:

```
> dotnet new --install .
  Restoring packages for C:\Users\mrahhal\.templateengine\dotnetcli\v1.1.0\scratch\restore.csproj...
C:\Program Files\dotnet\sdk\1.1.0\NuGet.targets(102,5): warning : The folder 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\BundlerMinifier.Core' contains an invalid version. [C:\Users\mrahhal\.templateengine\dotnetcli\v1.1.0\scratch\restore.csproj]
C:\Program Files\dotnet\sdk\1.1.0\NuGet.targets(102,5): warning : The folder 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\EntityFramework' contains an invalid version. [C:\Users\mrahhal\.templateengine\dotnetcli\v1.1.0\scratch\restore.csproj]
...
C:\Program Files\dotnet\sdk\1.1.0\NuGet.targets(102,5): warning : The folder 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\WindowsAzure.Storage' contains an invalid version. [C:\Users\mrahhal\.templateengine\dotnetcli\v1.1.0\scratch\restore.csproj]
C:\Users\mrahhal\.templateengine\dotnetcli\v1.1.0\scratch\restore.csproj : error NU1101: Unable to find package .. No packages exist with this id in source(s): Local Packages, Microsoft Visual Studio Offline Packages, nuget.org, TwistNuGet
  Generating MSBuild file C:\Users\mrahhal\.templateengine\dotnetcli\v1.1.0\scratch\obj\restore.csproj.nuget.g.props.
  Generating MSBuild file C:\Users\mrahhal\.templateengine\dotnetcli\v1.1.0\scratch\obj\restore.csproj.nuget.g.targets.
  Restore failed in 2.67 sec for C:\Users\mrahhal\.templateengine\dotnetcli\v1.1.0\scratch\restore.csproj.
```

I even tried packing a nuget package to try installing from the package instead, but it silently fails (no errors, just, nothing).

I thought about using a previous `dotnet` sdk version (by using global.json), but strangely enough the same output appeared.

I made sure that the project itself restores and builds successfully, and I was able to reproduce the output on multiple machines.

**Comment by neman** [comment]
Did you install .NET Core SDK 2.0.0? From your output lookls like you are using .NET Core SDK 1.1.0
`C:\Program Files\dotnet\sdk\1.1.0\NuGet.targets`

**Comment by mrahhal** [comment]
As I said, this particular invocation is when I thought the problem was related to SDK 2.0. So I created a global.json to go back to SDK 1.0. But the same problem occurred.

**Comment by mlorbetske** [comment]
Could you try `dotnet new --install .\` instead please? The trailing slash should be enough of a hint that it installs the directory instead of attempting to install a package from NuGet named `.`

**Comment by mrahhal** [comment]
Right, `dotnet new --install .\` worked. Could you expand on why this worked but `.` didn't? I'm pretty sure that I've always been using `dotnet new -i .` before SDK 2.0. But now I'm confused because I tried with SDK 1.0 and the same happened.

**Comment by mlorbetske** [comment]
In the 1.x CLI, we got a lot of feedback that people found the installation of NuGet packages confusing, the detection for package type installs was based on the presence of the "::" package ID/version separator in the parameter to the install switch. In 2.0, this has changed a bit to install the latest stable version of a package if only a package ID is specified. Instead of trying to define how a package ID must be structured, we look for some specific hints that the argument to `-i` [is a path on disk](https://github.com/dotnet/templating/blob/rel/2.1.0-preview1/src/Microsoft.TemplateEngine.Cli/Installer.cs#L36-L42). In this case, none of the "hints" are present, adding the slash changes that. We have discussed adding a check to see whether anything on disk matches the argument before attempting to use it as a package, but haven't gotten there just yet.
